lib/glific/profiles.ex

```diff
@@ -231,10 +231,46 @@ defmodule Glific.Profiles do
     end
   end
 
+  def handle_flow_action(:update_profile, context, action) do
+    attrs = %{
+      name: ContactField.parse_contact_field_value(context, action.value["name"]),
+      type: ContactField.parse_contact_field_value(context, action.value["type"])
+    }
+
+    case Repo.fetch_by(Profile, %{contact_id: context.contact.id}) do
+      {:ok, update_profile} ->
```

Comment:
What happens if a contact has multiple profiles

Reply:
It would update for the current profile

Reply:
I am not sure how is it fetching the current profile of the contact. This code seems to fetch any profile of the contact.